networkx/algorithms/link_analysis/hits_alg.py

```diff
@@ -363,23 +363,21 @@ def hits_scipy(G, max_iter=100, tol=1.0e-6, nstart=None, normalized=True):
 
     if len(G) == 0:
         return {}, {}
-    M = nx.to_scipy_sparse_matrix(G, nodelist=list(G))
-    (n, m) = M.shape  # should be square
-    A = M.T * M  # authority matrix
+    A = nx.to_scipy_sparse_matrix(G, nodelist=list(G))
+    (n, m) = A.shape  # should be square
+    ATA = A.T @ A  # authority matrix
     x = np.ones((n, 1)) / n  # initial guess
     # choose fixed starting vector if not given
-    if nstart is None:
-        x = np.ones((n, 1)) / n  # initial guess
-    else:
```

Comment:
It looks like if nstart is not None we set x=1/n and then go into the if statement where x gets set (and we wasted the setup time).
Would it be better to leave the if clause for when nstart is None and remove line 369 instead of removing the if clause?